### PR TITLE
Feature/db

### PR DESCRIPTION
**Pull request by magskai**
- Persist tweets & tweet images
- Fetch tweets & tweet images from db
- Serve tweets & their image hashes
- Check for <1 week old tweets in db and reprocess them if they have not been recently processed (newer tweets reprocessed more frequently)

DB changes:
- `tweet.last_processed` added
- `image.data` data type changed from `TEXT` to `MEDIUMTEXT` (large images were causing column out of range errors)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -177,6 +178,29 @@
         res.end();
     });
 
+    // Setup a request to get <1 week old tweets every 30 mins, and reprocess the tweets that have not been recently
+    // processed. Newer tweets are processed more frequently.
+    setInterval(async () => {
+        // 7 days in seconds
+        const maxAge = (60 * 60 * 24 * 7);
```
Comment (chris-dickson) [inline app.js]: Add this to the environment somewhere

File: app.js [hunk 2]
```diff
@@ -177,6 +178,29 @@
...
+        let tweets = await tweetApi.fetchRecent(maxAge);
+
+        if (tweets != null) {
+            const intervalRatio = 24;
```
Comment (chris-dickson) [inline app.js]: And this.

**Comment by magskai** [comment]
Changes pushed!

**Comment by dschroh** [comment]
Amazon RDS Aurora/MySQL 5.6
address: bringmehome.ck9bocwgsyz8.us-east-1.rds.amazonaws.com
username: homebringer
password: UpR6TwLHeYBpPnkhUGz[yzxhqKjkRrVy,FR4n
database: bringmehome

Note that the DB is only accessible from inside AWS (e.g. from Swarm) not from the Internet (e.g. our office).

**Pull request merged**